### ISSUE TITLE
False alarm in echo

### ISSUE THREAD

**Issue by s-p-k** [koalaman/shellcheck#274]
```sh
echo "-no" 
```
is treated like:
```sh
echo -n "o"
```
Which means that we get a warning like:
```sh
echo "-no"
     ^-- SC2039: In POSIX sh, echo flags are not supported.
```

This is a false alarm.

**Comment by ormaaj** [comment]
It isn't a false alarm. The warning could perhaps be more specific but it is true that you can't count on `echo` to behave a certain way when an option is specified.

I'd suggest:

> In POSIX sh, `echo` flags are only supported by shells that implement the XSI extensions. `printf` should almost always be preferred over `echo`'s `-e` and `-n`.

**Comment by s-p-k** [comment]
In the example above I'm not calling echo(1) with an option.
I'm just printing the "-no" string. Does POSIX disallow the '-' inside the string passed to echo(1) ?
I believe
``` sh
echo -n "o"
```
and
``` sh
echo "-no"
```
aren't the same thing. Please correct me if I'm wrong.

**Comment by koalaman** [comment]
`-no` is canonically the same as `-n -o`. In bash, it happens to print `-no` because `-o` is not a recognized option at this time.

If you instead tried `echo "-ne"`, you'd see that it doesn't print anything because both `-n` and `-e` are currently valid options.

I suggest you just avoid this mess and use `printf`.
